js_modules/dagit/packages/core/src/workspace/WorkspaceContext.tsx

```diff
@@ -189,10 +190,24 @@ const validateHiddenKeys = (parsed: unknown) => (Array.isArray(parsed) ? parsed
 const useVisibleRepos = (
   allRepos: DagsterRepoOption[],
 ): [DagsterRepoOption[], WorkspaceState['toggleVisible']] => {
-  const [hiddenKeys, setHiddenKeys] = useStateWithStorage<string[]>(
+  const {basePath} = React.useContext(AppContext);
+
+  const [oldHiddenKeys, setOldHiddenKeys] = useStateWithStorage<string[]>(
```

Comment:
This looks good! The only thing is I think we should audit our other uses of `useStateWithStorage` and see if there are really any that /should/ be shared across deployments? Maybe we push the basePath prefixing down into this hook and avoid sharing storage between deployments entirely...

Reply:
I'm not familiar with most of these callsites:
<img width="440" alt="Screen Shot 2022-04-11 at 3 28 28 PM" src="https://user-images.githubusercontent.com/2286579/162815091-c906fa9c-cd4f-42ed-bc18-b4040a69ce35.png">

But TimezoneContext seems like it doesn't have to be prefixed. Maybe we could an argument to `useStateWithStorage` to conditionally prefix for the callsites that need it.

Reply:
I think most of these are more specific to the user's preferences, and might be pretty annoying to have to set anew every time the user changes deployments (timezone, asset view, gantt mode, launchpad whitespace). Especially if you've got a lot of deployments in your org.